consensus/diffs.go

```diff
@@ -27,133 +28,85 @@ type FileContractDiff struct {
 // A SiafundOutputDiff indicates the addition or removal of a SiafundOutput in
 // the consensus set.
 type SiafundOutputDiff struct {
-	New           bool
+	Direction     DiffDirection
 	ID            SiafundOutputID
 	SiafundOutput SiafundOutput
 }
 
-// A SiafundPool diff contains the value of the siafundPool before the block
+// A SiafundPoolDiff contains the value of the siafundPool before the block
 // was applied, and after the block was applied. When applying the diff, set
-// siafundPool to `Adjusted`. When reversing the diff, set siafundPool to
-// `Previous`.
+// siafundPool to 'Adjusted'. When reverting the diff, set siafundPool to
+// 'Previous'.
 type SiafundPoolDiff struct {
 	Previous Currency
 	Adjusted Currency
 }
 
-// commitSiacoinOutputDiff takes a SiacoinOutputDiff and applies it to the
-// consensus set. `applied` indicates whether the diff is being applied or
-// reversed.
-func (s *State) commitSiacoinOutputDiff(scod SiacoinOutputDiff, applied bool) {
-	add := scod.New
-	if !applied {
-		add = !add
-	}
-
-	if add {
-		// Sanity check - output should not already exist.
-		if DEBUG {
-			_, exists := s.siacoinOutputs[scod.ID]
-			if exists {
-				panic("rogue new output in applyOutputDiff")
-			}
+// commitSiacoinOutputDiff applies or reverts a SiacoinOutputDiff.
+func (s *State) commitSiacoinOutputDiff(scod SiacoinOutputDiff, dir DiffDirection) {
+	// Sanity check - should not be adding an output twice, or deleting an
+	// output that does not exist.
+	if DEBUG {
+		_, exists := s.siacoinOutputs[scod.ID]
+		if exists == (scod.Direction == dir) {
+			panic("rogue output in commitSiacoinOutputDiff")
```

Comment:
rogue siacoin output

while we're at it. Not terribly important because the location of the panic gives you enough context even without any words.